src/DS/rds.c

```diff
@@ -601,6 +602,13 @@ static void ds_rds_add_xccdf_test_results(xmlDocPtr doc, xmlNodePtr reports,
 {
 	xmlNodePtr root_element = xmlDocGetRootElement(xccdf_result_file_doc);
 
+	if (root_element->ns && root_element->ns->href &&
```

Comment:
root_element is not getting checked to be non-NULL, if it's NULL this will lead to a NULL deref and crash in the best scenario.